packages/trpc/server/routers/viewer/workflows/util.ts

```diff
@@ -99,16 +95,80 @@ export const verifyEmailSender = async (email: string, userId: number, teamId: n
           teamId,
         },
       });
-    } else if (verifiedEmail.teamId !== teamId) {
+    } else if (teamId && verifiedEmail.teamId !== teamId) {
+      await prisma.verifiedEmail.create({
+        data: {
+          email,
+          userId,
+          teamId,
+        },
+      });
+    }
+    return;
+  }
+
+  const userEmail = await prisma.user.findFirst({
+    where: {
+      id: userId,
+      email,
+    },
+  });
+
+  if (userEmail) {
+    await prisma.verifiedEmail.create({
+      data: {
+        email,
+        userId,
+        teamId,
+      },
+    });
+    return;
+  }
+
+  if (teamId) {
+    const team = await prisma.team.findFirst({
+      where: {
+        id: teamId,
+      },
+      select: {
+        members: {
+          include: {
+            user: {
+              select: {
+                id: true,
+                email: true,
+              },
+            },
+          },
+        },
+      },
+    });
+
+    if (!team) {
+      throw new TRPCError({ code: "NOT_FOUND", message: "Team not found" });
+    }
+
+    const isTeamMember = team.members.some((member) => member.userId === userId);
+
+    if (!isTeamMember) {
+      throw new TRPCError({ code: "FORBIDDEN", message: "You are not a member of this team" });
+    }
+
+    const teamMemberEmail = team.members.filter((member) => member.user.email === email);
+
+    if (teamMemberEmail) {
       await prisma.verifiedEmail.create({
         data: {
           email,
           userId,
           teamId,
         },
       });
+      return;
     }
   }
+
+  throw new TRPCError({ code: "NOT_FOUND", message: "Email not verified" });
```

Comment:
No entry for `verifiedEmail` found and it's also not the user's email or any of the team members email